File: crates/rooch-finality/src/types/block.go
package types

type Block struct {
	BlockHash      string `json:"block_hash" description:"block hash"`
	BlockHeight    uint64 `json:"block_height" description:"block height"`
	BlockTimestamp uint64 `json:"block_timestamp" description:"block timestamp"`
}

type ChainSyncStatus struct {
	LatestBlockHeight               uint64 `json:"latest_block"`
	LatestBtcFinalizedBlockHeight   uint64 `json:"latest_btc_finalized_block"`
	EarliestBtcFinalizedBlockHeight uint64 `json:"earliest_btc_finalized_block"`
	LatestEthFinalizedBlockHeight   uint64 `json:"latest_eth_finalized_block"`
}

File: crates/rooch-finality/src/types/block.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Block {
    #[serde(rename = "block_hash")]
    pub block_hash: String,
    #[serde(rename = "block_height")]
    pub block_height: u64,
    #[serde(rename = "block_timestamp")]
    pub block_timestamp: u64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ChainSyncStatus {
    #[serde(rename = "latest_block")]
    pub latest_block_height: u64,
    #[serde(rename = "latest_btc_finalized_block")]
    pub latest_btc_finalized_block_height: u64,
    #[serde(rename = "earliest_btc_finalized_block")]
    pub earliest_btc_finalized_block_height: u64,
    #[serde(rename = "latest_eth_finalized_block")]
    pub latest_eth_finalized_block_height: u64,
} 